File: 2825_-_Make_String_A_Subsequence_Using_Cyclic_Increments/2825.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        str2_index = 0

        for c in str1:
            if (
                (c == str2[str2_index])
                or (c == "z" and str2[str2_index] == "a")
                or (ord(c) + 1) == ord(str2[str2_index])
            ):
                str2_index += 1
                if str2_index == len(str2):
                    return True
        return False

File: 2825_-_Make_String_A_Subsequence_Using_Cyclic_Increments/2825.cpp
class Solution {
  public:
    bool canMakeSubsequence(string str1, string str2) {
        int str1_len = str1.size();
        int str2_len = str2.size();
        if (str2_len > str1_len)
            return false;

        for (int i = 0, j = 0; i < str1_len; i++) {
            if (str1[i] == str2[j] || str1[i] + 1 == str2[j] ||
                (str1[i] == 'z' && str2[j] == 'a')) {
                ++j;
                if (j == str2_len) {
                    return true;
                }
            }
        }
        return false;
    }
};
